src/celeritas/track/SortTracksAction.hh

```diff
@@ -0,0 +1,56 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/track/SortTracksAction.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "celeritas/global/ActionInterface.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Sort track by status
+ *
+ * Provides Action interface to `celeritas::detail::partition_tracks_by_status`.
+ * 
+ * TODO: Use init params.init to configure sorting strategy
+ *
+ * \sa celeritas::detail::partition_tracks_by_status
+ */
+class SortTracksAction final : public ExplicitActionInterface
+{
+  public:
+    //! Construct with explicit Id
+    explicit SortTracksAction(ActionId id) : id_(id) {}
```

Comment:
Not necessary for this PR, but this would be the logical place to add a "strategy" enum value so that we can sort by status, particle type, etc.